src/components/styled-components/StyledDiv.js

```diff
@@ -159,6 +166,38 @@ const StyledDiv = styled.div`
 		margin: 1rem 0.5rem 1rem 0.5rem;
 		box-shadow: 0 0 1rem rgba(0, 0, 0, 0.25);
 	`}
+
+	${({ variant }) =>
+		variant === 'green-complex' &&
+		`
+		display: block;
+		width: 4rem;
+		height: 4rem;
+		border-radius: 50%;
+		background-color: var(--green-complex);
+		z-index: 9;
+	`}
+
+	${({ variant }) =>
+		variant === 'yellow-complex' &&
+		`
+		display: block;
+		width: 4rem;
+		height: 4rem;
+		border-radius: 50%;
+		background-color: var(--yellow-complex);
+		z-index: 9;
+	`}
+
+	${({ variant }) =>
+		variant === 'red-complex' &&
+		`
+		width: 4rem;
+		height: 4rem;
+		border-radius: 50%;
+		background-color: var(--red-complex);
+		z-index: 9;
+	`}
 `;
```

Comment:
Maybe it is better to create an additional component for this part of the StyledDiv (or StyledButton in this case) so you can specify all common styles between the variants like width, height, and border-radius and then only apply the background-color with conditional styling.